pkg/apiserver/debugapi/client.go

```diff
@@ -26,34 +28,64 @@ import (
 	"github.com/pingcap/tidb-dashboard/pkg/tidb"
 	"github.com/pingcap/tidb-dashboard/pkg/tiflash"
 	"github.com/pingcap/tidb-dashboard/pkg/tikv"
+	"github.com/pingcap/tidb-dashboard/pkg/utils/topology"
+	"github.com/thoas/go-funk"
 )
 
 const (
-	defaultTimeout = time.Second * 45 // Default profiling can be as long as 30s.
+	defaultTimeout = time.Second * 35 // Default profiling can be as long as 30s. Add 5 seconds for other overheads.
 )
 
+type Dispatcher struct {
```

Comment:
What is a Dispatcher? BTW I think it is not a good idea to abuse the Fx. For example, in this case seems that Dispatcher is totally internal. Then it should not be exposed to the outside.

Reply:
Maybe call it fetcher or add some notes to it? Yes, this should be injected via the parameter instead of Fx.